### ISSUE TITLE
[expo-image] Operation cancelled by user during querying the cache

### ISSUE THREAD

**Issue by BrodaNoel** [expo/expo#24125]
### Minimal reproducible example

https://snack.expo.dev/@brodanoel/expo-image-error

### Summary

`expo-image` is throwing an error, but the image is actually being properly render.

You will see 2 different errors.
The first time you open the Snack, you will see: `Operation cancelled by user during sending the request`, but then, change 1 line of code, so Expo Go refresh, and you'll see this another error: `Operation cancelled by user during querying the cache`

But, as you can see, the image is properly rendered.

This is kind of a big problem, because there is no way to know if the photo ACTUALLY rendered properly or not.

`"expo-image": "~1.3.2",`

I tested it and it's reproducible in real iOS device, and Expo Go (check snack). I didn't check Android

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.6.3
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.19.0 - ~/.nvm/versions/node/v16.19.0/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 8.19.3 - ~/.nvm/versions/node/v16.19.0/bin/npm
    Managers:
      CocoaPods: 1.11.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.10406996
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^49.0.7 => 49.0.7
      react: 18.2.0 => 18.2.0
      react-native: 0.72.3 => 0.72.3
    npmGlobalPackages:
      eas-cli: 4.1.2
      expo-cli: 6.3.10
    Expo Workflow: managed
```

**Comment by BrodaNoel** [comment]
In order to make it even harder to understand, I'm seen that sone times the photo is rendered and some times it isn't.

![image](https://github.com/expo/expo/assets/1837969/b2351c4b-1286-4cdd-9d8e-091d953a2c00)

![image](https://github.com/expo/expo/assets/1837969/80ceecd0-3696-47f9-b1ca-d3e0b2d83bb2)

**Comment by RRaideRR** [comment]
Yes - we're also suffering from this bug. I can confirm that this also occurs in `expo-image: 1.4.1`

**Comment by BrodaNoel** [comment]
Just to be clear: This is not a cosmetic issue. We had to rollback all the `expo-image` usage, until this gets fixed.

**Comment by RRaideRR** [comment]
I tend to agree. My app shows the user an error screen whenever the image loads with an error. This bug results in unnecessary error screens my users see. 

@tsapeta could you please give an estimation of the severity from your point of view? This would help me in understanding whether I also need to rollback to RN image.

**Comment by RRaideRR** [comment]
@BrodaNoel if I recall this correctly, the error was not always there. I think it came with one of the last versions. 

However, when I look for the message in `expo-image` I cannot find it in that package - so it seems it comes from one of the dependencies or underlying ios packages...?

**Comment by BrodaNoel** [comment]
I remember reading that error in some library related to WebP. I'm not sure. I can't check now. I'm away from my computer.

Are you also getting this error on WebP files?

I did a rollback to RN image

**Comment by BrodaNoel** [comment]
My photos are coming from Unolsahs, and they also are in AVIF...

**Comment by RRaideRR** [comment]
Okay I checked it. For both - "non-expired" images and for "expired" images - is no redirect taking place. Hmm - not quite sure where to go from this now. Pretty sure its related to SDWebImage but no clue how to go further from here.

**Comment by tsapeta** [comment]
> is it possible that you give an estimation of the severity from your point of view? This would help me in understanding whether I also need to rollback to RN image with the next release.

It seems to be a serious issue, so I'll make sure to prioritize that. I'm not sure if I can make it this week since I'll be on the conference, but I'll do my best 😅

**Comment by BrodaNoel** [comment]
@tsapeta fell free to mention me if in any moment you need somebody to test it. This is priority for my project so I can pause everything in order to help.

**Comment by fobos531** [comment]
> Just to be clear: This is not a cosmetic issue. We had to rollback all the `expo-image` usage, until this gets fixed.

Hey @BrodaNoel , I experienced this issue when I had an implementation of TabView where both screens where using expo-image. When I would swipe between them, this issue would occur. What worked for me as a workaround was to switch the cache policy of expo-image from the default to `memory-disk`.

**Comment by RRaideRR** [comment]
> > Just to be clear: This is not a cosmetic issue. We had to rollback all the `expo-image` usage, until this gets fixed.
> 
> Hey @BrodaNoel , I experienced this issue when I had an implementation of TabView where both screens were using expo-image. When I would swipe between them, this issue would occur. What worked for me as a temporary workaround was to switch the cache policy of expo-image from the default to `memory-disk`.

Interesting that this worked for you 🤔. The images in my app are always "memory-disk", so your approach unfortunately does not work in every scenario 😢

**Comment by BrodaNoel** [comment]
Yeah... that's weird. Because I switched to all the cache values and none of them worked. Or maybe I forgot to try that one.

Which expo-image version were you using?

**Comment by fobos531** [comment]
> Yeah... that's weird. Because I switched to all the cache values and none of them worked. Or maybe I forgot to try that one.
> 
> Which expo-image version were you using?

I am currently on 1.3.2

**Comment by BrodaNoel** [comment]
@tsapeta will this be released in a couple of days? Or is it already deployed?

**Comment by tsapeta** [comment]
@BrodaNoel Yes, I'll release it after the weekend 😉

**Comment by BrodaNoel** [comment]
![image](https://github.com/expo/expo/assets/1837969/1cdeafb3-6083-4d98-9d16-e69ff07de453)

Hahahaha

**Comment by RRaideRR** [comment]
@tsapeta thanks a lot for the quick fix 🤩. Really appreciate it 🙌🙏🙇🙇🙇

**Comment by tsapeta** [comment]
Published in `expo-image@1.5.1` 😉

**Comment by BrodaNoel** [comment]
@tsapeta I think this fix is the reason behind this issue: https://github.com/expo/expo/issues/24652

**Comment by BrodaNoel** [comment]
Fixed on SDK 50 ` "expo-image": "~1.10.6",`
